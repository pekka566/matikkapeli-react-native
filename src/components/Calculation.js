import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const Calculation = props => {
  const styles = {
    container: {
      flex: 0.1,
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: '1%',
      flexDirection: 'row'
    },
    text: {
      textAlign: 'center',
      flex: 1,
      fontWeight: 'bold',
      fontSize: 30,
      textShadowOffset: { width: 1, height: 1 },
      color: props.answer
        ? props.answer.isCorrect ? 'green' : 'red'
        : 'rgb(50, 50, 50)'
    }
  };
  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        {props.calculation.firstNumber + ' '}
        {props.calculation.operation}
        {' ' + props.calculation.secondNumber + ' = '}
        {props.answer ? props.answer.value : ''}
      </Text>
    </View>
  );
};

Calculation.propTypes = {
  calculation: PropTypes.object.isRequired,
  answer: PropTypes.object
};

export default Calculation;
