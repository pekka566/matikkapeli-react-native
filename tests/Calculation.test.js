import React from 'react';
import { shallow } from 'enzyme';
import Calculation from '../src/components/Calculation';

describe('<Calculation>', () => {
  let calculation = {
    firstNumber: 5,
    operation: '+',
    secondNumber: 6,
    result: 11
  };

  let calculation2 = {
    firstNumber: 15,
    operation: '+',
    secondNumber: 16,
    result: 31
  };
  const Component = <Calculation calculation={calculation}> </Calculation>;

  describe('Structure', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders calculation prop correctly', () => {
      const wrapper = shallow(Component);
      wrapper.setProps({ calculation: calculation2 });
      expect(wrapper).toMatchSnapshot();
    });

    it('renders props correctly', () => {
      const wrapper = shallow(Component);
      wrapper.setProps({ calculation: calculation2, answer: { value: 444 } });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
