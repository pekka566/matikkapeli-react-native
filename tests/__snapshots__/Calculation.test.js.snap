// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculation> Structure renders calculation prop correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": "1%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(50, 50, 50)",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "bold",
        "textAlign": "center",
        "textShadowOffset": Object {
          "height": 1,
          "width": 1,
        },
      }
    }
  >
    15 
    +
     16 = 
  </Text>
</View>
`;

exports[`<Calculation> Structure renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": "1%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(50, 50, 50)",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "bold",
        "textAlign": "center",
        "textShadowOffset": Object {
          "height": 1,
          "width": 1,
        },
      }
    }
  >
    5 
    +
     6 = 
  </Text>
</View>
`;

exports[`<Calculation> Structure renders props correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": "1%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "bold",
        "textAlign": "center",
        "textShadowOffset": Object {
          "height": 1,
          "width": 1,
        },
      }
    }
  >
    15 
    +
     16 = 
    444
  </Text>
</View>
`;
